diff -up cloud-init-22.1/cloudinit/config/cc_ntp.py.1~ cloud-init-22.1/cloudinit/config/cc_ntp.py
--- cloud-init-22.1/cloudinit/config/cc_ntp.py.1~	2022-03-02 15:27:35.480029682 +0100
+++ cloud-init-22.1/cloudinit/config/cc_ntp.py	2022-03-02 15:34:16.962126144 +0100
@@ -34,6 +34,7 @@ distros = [
     "fedora",
     "miraclelinux",
     "openEuler",
+    "openmandriva",
     "opensuse",
     "photon",
     "rhel",
@@ -96,6 +97,18 @@ DISTRO_CLIENT_CONFIG = {
             "confpath": "/etc/chrony/chrony.conf",
         },
     },
+    "openmandriva": {
+        "chrony": {
+            "service_name": "chronyd",
+        },
+        "ntp": {
+            "confpath": "/etc/ntp.conf",
+            "service_name": "ntpd",
+        },
+        "systemd-timesyncd": {
+            "check_exe": "/lib/systemd/systemd-timesyncd",
+        },
+    },
     "opensuse": {
         "chrony": {
             "service_name": "chronyd",
diff -up cloud-init-22.1/cloudinit/config/cc_yum_add_repo.py.1~ cloud-init-22.1/cloudinit/config/cc_yum_add_repo.py
--- cloud-init-22.1/cloudinit/config/cc_yum_add_repo.py.1~	2022-03-02 15:27:35.480029682 +0100
+++ cloud-init-22.1/cloudinit/config/cc_yum_add_repo.py	2022-03-02 15:35:00.051096230 +0100
@@ -19,7 +19,7 @@ entry, the config entry will be skipped.
 **Module frequency:** always
 
 **Supported distros:** almalinux, centos, cloudlinux, eurolinux, fedora,
-                       miraclelinux, openEuler, photon, rhel, rocky, virtuozzo
+                       miraclelinux, openEuler, openmandriva, photon, rhel, rocky, virtuozzo
 
 **Config keys**::
 
@@ -44,6 +44,7 @@ distros = [
     "eurolinux",
     "fedora",
     "openEuler",
+    "openmandriva",
     "photon",
     "rhel",
     "rocky",
diff -up cloud-init-22.1/cloudinit/distros/__init__.py.1~ cloud-init-22.1/cloudinit/distros/__init__.py
--- cloud-init-22.1/cloudinit/distros/__init__.py.1~	2022-03-02 15:27:35.481029662 +0100
+++ cloud-init-22.1/cloudinit/distros/__init__.py	2022-03-02 15:35:35.337245111 +0100
@@ -47,6 +47,7 @@ OSFAMILIES = {
         "fedora",
         "miraclelinux",
         "openEuler",
+        "openmandriva",
         "photon",
         "rhel",
         "rocky",
diff -up cloud-init-22.1/cloudinit/distros/openmandriva.py.1~ cloud-init-22.1/cloudinit/distros/openmandriva.py
--- cloud-init-22.1/cloudinit/distros/openmandriva.py.1~	2022-03-02 15:27:35.481029662 +0100
+++ cloud-init-22.1/cloudinit/distros/openmandriva.py	2022-03-02 15:27:35.481029662 +0100
@@ -0,0 +1,12 @@
+# Copyright (C) 2021 LinDev
+#
+# Author: Bernhard Rosenkraenzer <bero@lindev.ch>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+from cloudinit.distros import fedora
+
+class Distro(fedora.Distro):
+    pass
+
+# vi: ts=4 expandtab
diff -up cloud-init-22.1/cloudinit/net/sysconfig.py.1~ cloud-init-22.1/cloudinit/net/sysconfig.py
--- cloud-init-22.1/cloudinit/net/sysconfig.py.1~	2022-03-02 15:27:35.481029662 +0100
+++ cloud-init-22.1/cloudinit/net/sysconfig.py	2022-03-02 15:36:01.192617356 +0100
@@ -29,6 +29,7 @@ KNOWN_DISTROS = [
     "fedora",
     "miraclelinux",
     "openEuler",
+    "openmandriva",
     "rhel",
     "rocky",
     "suse",
diff -up cloud-init-22.1/cloudinit/util.py.1~ cloud-init-22.1/cloudinit/util.py
--- cloud-init-22.1/cloudinit/util.py.1~	2022-03-02 15:27:35.482029642 +0100
+++ cloud-init-22.1/cloudinit/util.py	2022-03-02 15:37:29.939438635 +0100
@@ -585,6 +585,7 @@ def _get_variant(info):
             "fedora",
             "miraclelinux",
             "openeuler",
+            "openmandriva",
             "photon",
             "rhel",
             "rocky",
diff -up cloud-init-22.1/config/cloud.cfg.tmpl.1~ cloud-init-22.1/config/cloud.cfg.tmpl
--- cloud-init-22.1/config/cloud.cfg.tmpl.1~	2022-02-15 19:59:19.000000000 +0100
+++ cloud-init-22.1/config/cloud.cfg.tmpl	2022-03-02 15:39:37.616248314 +0100
@@ -33,7 +33,7 @@ disable_root: true
 {% endif %}
 
 {% if variant in ["almalinux", "alpine", "amazon", "centos", "cloudlinux", "eurolinux",
-                  "fedora", "miraclelinux", "openEuler", "rhel", "rocky", "virtuozzo"] %}
+                  "fedora", "miraclelinux", "openEuler", "openmandriva", "rhel", "rocky", "virtuozzo"] %}
 mount_default_fields: [~, ~, 'auto', 'defaults,nofail', '0', '2']
 {% if variant == "amazon" %}
 resize_rootfs: noblock
@@ -111,8 +111,8 @@ cloud_config_modules:
  - locale
 {% endif %}
  - set-passwords
-{% if variant in ["rhel", "fedora", "photon"] %}
-{% if variant not in ["photon"] %}
+{% if variant in ["rhel", "fedora", "photon", "openmandriva"] %}
+{% if variant not in ["photon", "openmandriva"] %}
  - spacewalk
 {% endif %}
  - yum-add-repo
@@ -178,7 +178,7 @@ system_info:
    # This will affect which distro class gets used
 {% if variant in ["almalinux", "alpine", "amazon", "arch", "centos", "cloudlinux", "debian",
                   "eurolinux", "fedora", "freebsd", "gentoo", "netbsd", "miraclelinux", "openbsd", "openEuler",
-                  "photon", "rhel", "rocky", "suse", "ubuntu", "virtuozzo"] %}
+                  "openmandriva", "photon", "rhel", "rocky", "suse", "ubuntu", "virtuozzo"] %}
    distro: {{ variant }}
 {% elif variant in ["dragonfly"] %}
    distro: dragonflybsd
@@ -232,7 +232,7 @@ system_info:
          security: http://ports.ubuntu.com/ubuntu-ports
    ssh_svcname: ssh
 {% elif variant in ["almalinux", "alpine", "amazon", "arch", "centos", "cloudlinux", "eurolinux",
-                    "fedora", "gentoo", "miraclelinux", "openEuler", "rhel", "rocky", "suse", "virtuozzo"] %}
+                    "fedora", "gentoo", "miraclelinux", "openEuler", "openmandriva", "rhel", "rocky", "suse", "virtuozzo"] %}
    # Default user name + that default users groups (if added/used)
    default_user:
 {% if variant == "amazon" %}
@@ -254,6 +254,8 @@ system_info:
      groups: [adm, sudo]
 {% elif variant == "arch" %}
      groups: [wheel, users]
+{% elif variant == "openmandriva" %}
+     groups: [wheel, users]
 {% else %}
      groups: [wheel, adm, systemd-journal]
 {% endif %}
diff -up cloud-init-22.1/systemd/cloud-init-generator.tmpl.1~ cloud-init-22.1/systemd/cloud-init-generator.tmpl
--- cloud-init-22.1/systemd/cloud-init-generator.tmpl.1~	2022-03-02 15:27:35.482029642 +0100
+++ cloud-init-22.1/systemd/cloud-init-generator.tmpl	2022-03-02 15:40:30.887900863 +0100
@@ -84,7 +84,7 @@ default() {
 check_for_datasource() {
     local ds_rc=""
 {% if variant in ["almalinux", "centos", "cloudlinux", "eurolinux", "fedora", 
-                  "miraclelinux", "openEuler", "rhel", "rocky", "virtuozzo"] %}
+                  "miraclelinux", "openEuler", "openmandriva", "rhel", "rocky", "virtuozzo"] %}
     local dsidentify="/usr/libexec/cloud-init/ds-identify"
 {% else %}
     local dsidentify="/usr/lib/cloud-init/ds-identify"
diff -up cloud-init-22.1/systemd/cloud-init.service.tmpl.1~ cloud-init-22.1/systemd/cloud-init.service.tmpl
--- cloud-init-22.1/systemd/cloud-init.service.tmpl.1~	2022-03-02 15:27:35.483029622 +0100
+++ cloud-init-22.1/systemd/cloud-init.service.tmpl	2022-03-02 15:40:49.936417006 +0100
@@ -13,7 +13,7 @@ After=systemd-networkd-wait-online.servi
 After=networking.service
 {% endif %}
 {% if variant in ["almalinux", "centos", "cloudlinux", "eurolinux", "fedora",
-                  "miraclelinux", "openEuler", "rhel", "rocky", "virtuozzo"] %}
+                  "miraclelinux", "openEuler", "openmandriva", "rhel", "rocky", "virtuozzo"] %}
 After=network.service
 After=NetworkManager.service
 {% endif %}
diff -up cloud-init-22.1/tools/render-cloudcfg.1~ cloud-init-22.1/tools/render-cloudcfg
--- cloud-init-22.1/tools/render-cloudcfg.1~	2022-03-02 15:27:35.483029622 +0100
+++ cloud-init-22.1/tools/render-cloudcfg	2022-03-02 15:41:06.015007801 +0100
@@ -25,6 +25,7 @@ def main():
         "netbsd",
         "openbsd",
         "openEuler",
+        "openmandriva",
         "photon",
         "rhel",
         "suse",
