diff -up cloud-init-22.2/cloudinit/config/cc_ntp.py.1~ cloud-init-22.2/cloudinit/config/cc_ntp.py
--- cloud-init-22.2/cloudinit/config/cc_ntp.py.1~	2022-05-18 18:04:36.000000000 +0200
+++ cloud-init-22.2/cloudinit/config/cc_ntp.py	2022-06-01 14:20:07.941923436 +0200
@@ -30,6 +30,7 @@ distros = [
     "fedora",
     "miraclelinux",
     "openEuler",
+    "openmandriva",
     "opensuse",
     "photon",
     "rhel",
@@ -92,6 +93,18 @@ DISTRO_CLIENT_CONFIG = {
             "confpath": "/etc/chrony/chrony.conf",
         },
     },
+    "openmandriva": {
+        "chrony": {
+            "service_name": "chronyd",
+        },
+        "ntp": {
+            "confpath": "/etc/ntp.conf",
+            "service_name": "ntpd",
+        },
+        "systemd-timesyncd": {
+            "check_exe": "/lib/systemd/systemd-timesyncd",
+        },
+    },
     "opensuse": {
         "chrony": {
             "service_name": "chronyd",
diff -up cloud-init-22.2/cloudinit/config/cc_yum_add_repo.py.1~ cloud-init-22.2/cloudinit/config/cc_yum_add_repo.py
--- cloud-init-22.2/cloudinit/config/cc_yum_add_repo.py.1~	2022-05-18 18:04:36.000000000 +0200
+++ cloud-init-22.2/cloudinit/config/cc_yum_add_repo.py	2022-06-01 14:20:07.941923436 +0200
@@ -29,6 +29,7 @@ distros = [
     "eurolinux",
     "fedora",
     "openEuler",
+    "openmandriva",
     "photon",
     "rhel",
     "rocky",
diff -up cloud-init-22.2/cloudinit/distros/__init__.py.1~ cloud-init-22.2/cloudinit/distros/__init__.py
--- cloud-init-22.2/cloudinit/distros/__init__.py.1~	2022-05-18 18:04:36.000000000 +0200
+++ cloud-init-22.2/cloudinit/distros/__init__.py	2022-06-01 14:20:07.941923436 +0200
@@ -47,6 +47,7 @@ OSFAMILIES = {
         "fedora",
         "miraclelinux",
         "openEuler",
+        "openmandriva",
         "photon",
         "rhel",
         "rocky",
diff -up cloud-init-22.2/cloudinit/distros/openmandriva.py.1~ cloud-init-22.2/cloudinit/distros/openmandriva.py
--- cloud-init-22.2/cloudinit/distros/openmandriva.py.1~	2022-06-01 14:20:07.941923436 +0200
+++ cloud-init-22.2/cloudinit/distros/openmandriva.py	2022-06-01 14:20:07.941923436 +0200
@@ -0,0 +1,12 @@
+# Copyright (C) 2022 LinDev
+#
+# Author: Bernhard Rosenkraenzer <bero@lindev.ch>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+from cloudinit.distros import fedora
+
+class Distro(fedora.Distro):
+    pass
+
+# vi: ts=4 expandtab
diff -up cloud-init-22.2/cloudinit/net/sysconfig.py.1~ cloud-init-22.2/cloudinit/net/sysconfig.py
--- cloud-init-22.2/cloudinit/net/sysconfig.py.1~	2022-05-18 18:04:36.000000000 +0200
+++ cloud-init-22.2/cloudinit/net/sysconfig.py	2022-06-01 14:20:07.941923436 +0200
@@ -28,6 +28,7 @@ KNOWN_DISTROS = [
     "fedora",
     "miraclelinux",
     "openEuler",
+    "openmandriva",
     "rhel",
     "rocky",
     "suse",
diff -up cloud-init-22.2/cloudinit/util.py.1~ cloud-init-22.2/cloudinit/util.py
--- cloud-init-22.2/cloudinit/util.py.1~	2022-05-18 18:04:36.000000000 +0200
+++ cloud-init-22.2/cloudinit/util.py	2022-06-01 14:20:07.941923436 +0200
@@ -601,6 +601,7 @@ def _get_variant(info):
             "fedora",
             "miraclelinux",
             "openeuler",
+            "openmandriva",
             "photon",
             "rhel",
             "rocky",
diff -up cloud-init-22.2/config/cloud.cfg.tmpl.1~ cloud-init-22.2/config/cloud.cfg.tmpl
--- cloud-init-22.2/config/cloud.cfg.tmpl.1~	2022-05-18 18:04:36.000000000 +0200
+++ cloud-init-22.2/config/cloud.cfg.tmpl	2022-06-01 14:26:15.811150749 +0200
@@ -33,7 +33,7 @@ disable_root: true
 {% endif %}
 
 {% if variant in ["almalinux", "alpine", "amazon", "centos", "cloudlinux", "eurolinux",
-                  "fedora", "miraclelinux", "openEuler", "rhel", "rocky", "virtuozzo"] %}
+                  "fedora", "miraclelinux", "openEuler", "openmandriva", "rhel", "rocky", "virtuozzo"] %}
 {% if variant == "rhel" %}
 mount_default_fields: [~, ~, 'auto', 'defaults,nofail,x-systemd.requires=cloud-init.service,_netdev', '0', '2']
 {% else %}
@@ -125,8 +125,8 @@ cloud_config_modules:
 {% if variant in ["rhel"] %}
  - rh_subscription
 {% endif %}
-{% if variant in ["rhel", "fedora", "photon"] %}
-{% if variant not in ["photon"] %}
+{% if variant in ["rhel", "fedora", "openmandriva", "photon"] %}
+{% if variant not in ["openmandriva", "photon"] %}
  - spacewalk
 {% endif %}
  - yum-add-repo
@@ -192,7 +192,7 @@ system_info:
    # This will affect which distro class gets used
 {% if variant in ["almalinux", "alpine", "amazon", "arch", "centos", "cloudlinux", "debian",
                   "eurolinux", "fedora", "freebsd", "gentoo", "netbsd", "miraclelinux", "openbsd", "openEuler",
-                  "photon", "rhel", "rocky", "suse", "ubuntu", "virtuozzo"] %}
+                  "openmandriva", "photon", "rhel", "rocky", "suse", "ubuntu", "virtuozzo"] %}
    distro: {{ variant }}
 {% elif variant in ["dragonfly"] %}
    distro: dragonflybsd
@@ -245,7 +245,7 @@ system_info:
          security: http://ports.ubuntu.com/ubuntu-ports
    ssh_svcname: ssh
 {% elif variant in ["almalinux", "alpine", "amazon", "arch", "centos", "cloudlinux", "eurolinux",
-                    "fedora", "gentoo", "miraclelinux", "openEuler", "rhel", "rocky", "suse", "virtuozzo"] %}
+                    "fedora", "gentoo", "miraclelinux", "openEuler", "openmandriva", "rhel", "rocky", "suse", "virtuozzo"] %}
    # Default user name + that default users groups (if added/used)
    default_user:
 {% if variant == "amazon" %}
@@ -271,6 +271,8 @@ system_info:
      groups: [adm, sudo]
 {% elif variant == "arch" %}
      groups: [wheel, users]
+{% elif variant == "openmandriva" %}
+     groups: [wheel, users, systemd-journal]
 {% elif variant == "rhel" %}
      groups: [adm, systemd-journal]
 {% else %}
@@ -349,4 +351,7 @@ system_info:
 {% elif variant in ["dragonfly"] %}
    network:
       renderers: ['freebsd']
+{% elif variant == "openmandriva" %}
+   network:
+      renderers: ['network-manager', 'networkd']
 {% endif %}
diff -up cloud-init-22.2/systemd/cloud-init-generator.tmpl.1~ cloud-init-22.2/systemd/cloud-init-generator.tmpl
--- cloud-init-22.2/systemd/cloud-init-generator.tmpl.1~	2022-06-01 14:20:07.942923440 +0200
+++ cloud-init-22.2/systemd/cloud-init-generator.tmpl	2022-06-01 14:26:54.896283822 +0200
@@ -21,7 +21,7 @@ CLOUD_SYSTEM_TARGET="/usr/lib/systemd/sy
 CLOUD_SYSTEM_TARGET="/lib/systemd/system/cloud-init.target"
 {% endif %}
 {% if variant in ["almalinux", "centos", "cloudlinux", "eurolinux", "fedora",
-                  "miraclelinux", "openEuler", "rhel", "rocky", "virtuozzo"] %}
+                  "miraclelinux", "openEuler", "openmandriva", "rhel", "rocky", "virtuozzo"] %}
     dsidentify="/usr/libexec/cloud-init/ds-identify"
 {% else %}
     dsidentify="/usr/lib/cloud-init/ds-identify"
diff -up cloud-init-22.2/systemd/cloud-init.service.tmpl.1~ cloud-init-22.2/systemd/cloud-init.service.tmpl
--- cloud-init-22.2/systemd/cloud-init.service.tmpl.1~	2022-05-18 18:04:36.000000000 +0200
+++ cloud-init-22.2/systemd/cloud-init.service.tmpl	2022-06-01 14:20:07.942923440 +0200
@@ -13,7 +13,7 @@ After=systemd-networkd-wait-online.servi
 After=networking.service
 {% endif %}
 {% if variant in ["almalinux", "centos", "cloudlinux", "eurolinux", "fedora",
-                  "miraclelinux", "openEuler", "rhel", "rocky", "virtuozzo"] %}
+                  "miraclelinux", "openEuler", "openmandriva", "rhel", "rocky", "virtuozzo"] %}
 After=network.service
 After=NetworkManager.service
 {% endif %}
diff -up cloud-init-22.2/tools/render-cloudcfg.1~ cloud-init-22.2/tools/render-cloudcfg
--- cloud-init-22.2/tools/render-cloudcfg.1~	2022-05-18 18:04:36.000000000 +0200
+++ cloud-init-22.2/tools/render-cloudcfg	2022-06-01 14:20:07.942923440 +0200
@@ -25,6 +25,7 @@ def main():
         "netbsd",
         "openbsd",
         "openEuler",
+        "openmandriva",
         "photon",
         "rhel",
         "suse",
