diff -up cloud-init-21.4/config/cloud.cfg.tmpl.omv~ cloud-init-21.4/config/cloud.cfg.tmpl
--- cloud-init-21.4/config/cloud.cfg.tmpl.omv~	2021-12-16 14:05:23.410727624 +0100
+++ cloud-init-21.4/config/cloud.cfg.tmpl	2021-12-16 14:06:37.736998781 +0100
@@ -269,6 +269,10 @@ system_info:
    paths:
       cloud_dir: /var/lib/cloud/
       templates_dir: /etc/cloud/templates/
+{% if variant == "openmandriva" %}
+   network:
+      renderers: ['networkd']
+{% endif %}
    ssh_svcname: sshd
 {% elif variant in ["freebsd"] %}
    # Default user name + that default users groups (if added/used)
diff -up cloud-init-21.4/systemd/cloud-init-local.service.tmpl.omv~ cloud-init-21.4/systemd/cloud-init-local.service.tmpl
diff -up cloud-init-21.4/systemd/cloud-init.service.tmpl.omv~ cloud-init-21.4/systemd/cloud-init.service.tmpl
--- cloud-init-21.4/systemd/cloud-init.service.tmpl.omv~	2021-12-16 14:09:32.966645277 +0100
+++ cloud-init-21.4/systemd/cloud-init.service.tmpl	2021-12-16 14:10:32.640867328 +0100
@@ -13,10 +13,14 @@ After=systemd-networkd-wait-online.servi
 After=networking.service
 {% endif %}
 {% if variant in ["almalinux", "centos", "cloudlinux", "eurolinux", "fedora",
-                  "openEuler", "openmandriva", "rhel", "rocky", "virtuozzo"] %}
+                  "openEuler", "rhel", "rocky", "virtuozzo"] %}
 After=network.service
 After=NetworkManager.service
 {% endif %}
+{% if variant in ["openmandriva"] %}
+After=network.service
+After=systemd-networkd.service
+{% endif %}
 {% if variant in ["suse"] %}
 After=wicked.service
 # setting hostname via hostnamectl depends on dbus, which otherwise
