--- cloud-init-26.2/cloudinit/sources/DataSourceCloudStack.py.omv~	2026-08-02 16:09:12.243799559 +0200
+++ cloud-init-26.2/cloudinit/sources/DataSourceCloudStack.py	2026-08-02 16:09:12.288712877 +0200
@@ -301,6 +301,10 @@ def get_data_server():
     except gaierror:
         LOG.debug("DNS Entry data-server not found")
         return None
+    except Exception as e:
+        # getaddrinfo can fail with exceptions other than gaierror
+        LOG.debug("Attempt to look up DNS for data-server failed: %s", str(e))
+        return None
     else:
         return addrinfo[0][4][0]  # return IP
 
