diff -up cloud-init-21.4/cloudinit/config/cc_ntp.py.1~ cloud-init-21.4/cloudinit/config/cc_ntp.py
--- cloud-init-21.4/cloudinit/config/cc_ntp.py.1~	2021-11-02 20:35:08.000000000 +0100
+++ cloud-init-21.4/cloudinit/config/cc_ntp.py	2021-12-04 22:51:12.528406998 +0100
@@ -25,8 +25,8 @@ frequency = PER_INSTANCE
 NTP_CONF = '/etc/ntp.conf'
 NR_POOL_SERVERS = 4
 distros = ['almalinux', 'alpine', 'centos', 'cloudlinux', 'debian',
-           'eurolinux', 'fedora', 'openEuler', 'opensuse', 'photon',
-           'rhel', 'rocky', 'sles', 'ubuntu', 'virtuozzo']
+           'eurolinux', 'fedora', 'openEuler', 'openmandriva', 'opensuse',
+           'photon', 'rhel', 'rocky', 'sles', 'ubuntu', 'virtuozzo']
 
 NTP_CLIENT_CONFIG = {
     'chrony': {
@@ -81,6 +81,11 @@ DISTRO_CLIENT_CONFIG = {
             'confpath': '/etc/chrony/chrony.conf',
         },
     },
+    'openmandriva': {
+        'systemd-timesyncd': {
+            'check_exe': '/lib/systemd/systemd-timesyncd',
+        },
+    },
     'opensuse': {
         'chrony': {
             'service_name': 'chronyd',
diff -up cloud-init-21.4/cloudinit/config/cc_yum_add_repo.py.1~ cloud-init-21.4/cloudinit/config/cc_yum_add_repo.py
--- cloud-init-21.4/cloudinit/config/cc_yum_add_repo.py.1~	2021-11-02 20:35:08.000000000 +0100
+++ cloud-init-21.4/cloudinit/config/cc_yum_add_repo.py	2021-12-04 22:51:12.528406998 +0100
@@ -19,7 +19,7 @@ entry, the config entry will be skipped.
 **Module frequency:** always
 
 **Supported distros:** almalinux, centos, cloudlinux, eurolinux, fedora,
-                       openEuler, photon, rhel, rocky, virtuozzo
+                       openEuler, openmandriva, photon, rhel, rocky, virtuozzo
 
 **Config keys**::
 
@@ -38,7 +38,7 @@ from configparser import ConfigParser
 from cloudinit import util
 
 distros = ['almalinux', 'centos', 'cloudlinux', 'eurolinux', 'fedora',
-           'openEuler', 'photon', 'rhel', 'rocky', 'virtuozzo']
+           'openEuler', 'openmandriva', 'photon', 'rhel', 'rocky', 'virtuozzo']
 
 
 def _canonicalize_id(repo_id):
diff -up cloud-init-21.4/cloudinit/distros/__init__.py.1~ cloud-init-21.4/cloudinit/distros/__init__.py
--- cloud-init-21.4/cloudinit/distros/__init__.py.1~	2021-11-02 20:35:08.000000000 +0100
+++ cloud-init-21.4/cloudinit/distros/__init__.py	2021-12-04 22:51:12.529407003 +0100
@@ -50,7 +50,8 @@ OSFAMILIES = {
     'freebsd': ['freebsd'],
     'gentoo': ['gentoo'],
     'redhat': ['almalinux', 'amazon', 'centos', 'cloudlinux', 'eurolinux',
-               'fedora', 'openEuler', 'photon', 'rhel', 'rocky', 'virtuozzo'],
+               'fedora', 'openEuler', 'openmandriva', 'photon', 'rhel',
+               'rocky', 'virtuozzo'],
     'suse': ['opensuse', 'sles'],
 }
 
diff -up cloud-init-21.4/cloudinit/distros/openmandriva.py.1~ cloud-init-21.4/cloudinit/distros/openmandriva.py
--- cloud-init-21.4/cloudinit/distros/openmandriva.py.1~	2021-12-04 22:51:12.529407003 +0100
+++ cloud-init-21.4/cloudinit/distros/openmandriva.py	2021-12-04 22:51:12.529407003 +0100
@@ -0,0 +1,12 @@
+# Copyright (C) 2021 LinDev
+#
+# Author: Bernhard Rosenkraenzer <bero@lindev.ch>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+from cloudinit.distros import ubuntu
+
+class Distro(ubuntu.Distro):
+    pass
+
+# vi: ts=4 expandtab
diff -up cloud-init-21.4/cloudinit/net/sysconfig.py.1~ cloud-init-21.4/cloudinit/net/sysconfig.py
--- cloud-init-21.4/cloudinit/net/sysconfig.py.1~	2021-11-02 20:35:08.000000000 +0100
+++ cloud-init-21.4/cloudinit/net/sysconfig.py	2021-12-04 22:51:12.529407003 +0100
@@ -19,7 +19,7 @@ from .network_state import (
 
 LOG = logging.getLogger(__name__)
 KNOWN_DISTROS = ['almalinux', 'centos', 'cloudlinux', 'eurolinux', 'fedora',
-                 'openEuler', 'rhel', 'rocky', 'suse', 'virtuozzo']
+                 'openEuler', 'openmandriva', 'rhel', 'rocky', 'suse', 'virtuozzo']
 NM_CFG_FILE = "/etc/NetworkManager/NetworkManager.conf"
 
 
diff -up cloud-init-21.4/cloudinit/util.py.1~ cloud-init-21.4/cloudinit/util.py
--- cloud-init-21.4/cloudinit/util.py.1~	2021-11-02 20:35:08.000000000 +0100
+++ cloud-init-21.4/cloudinit/util.py	2021-12-04 22:51:12.530407009 +0100
@@ -549,8 +549,8 @@ def system_info():
         linux_dist = info['dist'][0].lower()
         if linux_dist in (
                 'almalinux', 'alpine', 'arch', 'centos', 'cloudlinux',
-                'debian', 'eurolinux', 'fedora', 'openEuler', 'photon',
-                'rhel', 'rocky', 'suse', 'virtuozzo'):
+                'debian', 'eurolinux', 'fedora', 'openEuler', 'openmandriva',
+                'photon', 'rhel', 'rocky', 'suse', 'virtuozzo'):
             var = linux_dist
         elif linux_dist in ('ubuntu', 'linuxmint', 'mint'):
             var = 'ubuntu'
diff -up cloud-init-21.4/config/cloud.cfg.tmpl.1~ cloud-init-21.4/config/cloud.cfg.tmpl
--- cloud-init-21.4/config/cloud.cfg.tmpl.1~	2021-11-02 20:35:08.000000000 +0100
+++ cloud-init-21.4/config/cloud.cfg.tmpl	2021-12-04 22:57:27.571647992 +0100
@@ -33,7 +33,7 @@ disable_root: true
 {% endif %}
 
 {% if variant in ["almalinux", "alpine", "amazon", "centos", "cloudlinux", "eurolinux",
-                  "fedora", "openEuler", "rhel", "rocky", "virtuozzo"] %}
+                  "fedora", "openEuler", "openmandriva", "rhel", "rocky", "virtuozzo"] %}
 mount_default_fields: [~, ~, 'auto', 'defaults,nofail', '0', '2']
 {% if variant == "amazon" %}
 resize_rootfs: noblock
@@ -110,8 +110,8 @@ cloud_config_modules:
  - locale
 {% endif %}
  - set-passwords
-{% if variant in ["rhel", "fedora", "photon"] %}
-{% if variant not in ["photon"] %}
+{% if variant in ["rhel", "fedora", "photon", "openmandriva"] %}
+{% if variant not in ["photon", "openmandriva"] %}
  - spacewalk
 {% endif %}
  - yum-add-repo
@@ -177,7 +177,7 @@ system_info:
    # This will affect which distro class gets used
 {% if variant in ["almalinux", "alpine", "amazon", "arch", "centos", "cloudlinux", "debian",
                   "eurolinux", "fedora", "freebsd", "netbsd", "openbsd", "openEuler",
-                  "photon", "rhel", "rocky", "suse", "ubuntu", "virtuozzo"] %}
+                  "openmandriva", "photon", "rhel", "rocky", "suse", "ubuntu", "virtuozzo"] %}
    distro: {{ variant }}
 {% elif variant in ["dragonfly"] %}
    distro: dragonflybsd
@@ -231,13 +231,18 @@ system_info:
          security: http://ports.ubuntu.com/ubuntu-ports
    ssh_svcname: ssh
 {% elif variant in ["almalinux", "alpine", "amazon", "arch", "centos", "cloudlinux", "eurolinux",
-                    "fedora", "openEuler", "rhel", "rocky", "suse", "virtuozzo"] %}
+                    "fedora", "openEuler", "openmandriva", "rhel", "rocky", "suse", "virtuozzo"] %}
    # Default user name + that default users groups (if added/used)
    default_user:
 {% if variant == "amazon" %}
      name: ec2-user
      lock_passwd: True
      gecos: EC2 Default User
+{% elif variant == "openmandriva" %}
+     name: omv
+     lock_passwd: True
+     gecos: OpenMandriva User
+     shell: /bin/bash
 {% else %}
      name: {{ variant }}
      lock_passwd: True
@@ -249,6 +254,8 @@ system_info:
      groups: [adm, sudo]
 {% elif variant == "arch" %}
      groups: [wheel, users]
+{% elif variant == "openmandriva" %}
+     groups: [wheel, users]
 {% else %}
      groups: [wheel, adm, systemd-journal]
 {% endif %}
diff -up cloud-init-21.4/systemd/cloud-init-generator.tmpl.1~ cloud-init-21.4/systemd/cloud-init-generator.tmpl
--- cloud-init-21.4/systemd/cloud-init-generator.tmpl.1~	2021-11-02 20:35:08.000000000 +0100
+++ cloud-init-21.4/systemd/cloud-init-generator.tmpl	2021-12-04 22:51:12.530407009 +0100
@@ -84,7 +84,7 @@ default() {
 check_for_datasource() {
     local ds_rc=""
 {% if variant in ["almalinux", "centos", "cloudlinux", "eurolinux", "fedora", 
-                  "openEuler", "rhel", "rocky", "virtuozzo"] %}
+                  "openEuler", "openmandriva", "rhel", "rocky", "virtuozzo"] %}
     local dsidentify="/usr/libexec/cloud-init/ds-identify"
 {% else %}
     local dsidentify="/usr/lib/cloud-init/ds-identify"
diff -up cloud-init-21.4/systemd/cloud-init.service.tmpl.1~ cloud-init-21.4/systemd/cloud-init.service.tmpl
--- cloud-init-21.4/systemd/cloud-init.service.tmpl.1~	2021-11-02 20:35:08.000000000 +0100
+++ cloud-init-21.4/systemd/cloud-init.service.tmpl	2021-12-04 22:51:12.530407009 +0100
@@ -13,7 +13,7 @@ After=systemd-networkd-wait-online.servi
 After=networking.service
 {% endif %}
 {% if variant in ["almalinux", "centos", "cloudlinux", "eurolinux", "fedora",
-                  "openEuler", "rhel", "rocky", "virtuozzo"] %}
+                  "openEuler", "openmandriva", "rhel", "rocky", "virtuozzo"] %}
 After=network.service
 After=NetworkManager.service
 {% endif %}
diff -up cloud-init-21.4/tools/render-cloudcfg.1~ cloud-init-21.4/tools/render-cloudcfg
--- cloud-init-21.4/tools/render-cloudcfg.1~	2021-11-02 20:35:08.000000000 +0100
+++ cloud-init-21.4/tools/render-cloudcfg	2021-12-04 22:51:12.530407009 +0100
@@ -5,8 +5,8 @@ import os
 import sys
 
 VARIANTS = ["almalinux", "alpine", "amazon", "arch", "centos", "cloudlinux", "debian",
-            "eurolinux", "fedora", "freebsd", "netbsd", "openbsd", "openEuler", "photon",
-            "rhel", "suse","rocky", "ubuntu", "unknown", "virtuozzo"]
+            "eurolinux", "fedora", "freebsd", "netbsd", "openbsd", "openEuler", "openmandriva",
+            "photon", "rhel", "suse","rocky", "ubuntu", "unknown", "virtuozzo"]
 
 
 if "avoid-pep8-E402-import-not-top-of-file":
