diff -up cloud-init-21.4/cloudinit/distros/openmandriva.py.omv~ cloud-init-21.4/cloudinit/distros/openmandriva.py
--- cloud-init-21.4/cloudinit/distros/openmandriva.py.omv~	2021-12-17 00:39:55.065326982 +0100
+++ cloud-init-21.4/cloudinit/distros/openmandriva.py	2021-12-17 00:40:06.433362919 +0100
@@ -4,9 +4,9 @@
 #
 # This file is part of cloud-init. See LICENSE file for license information.
 
-from cloudinit.distros import fedora
+from cloudinit.distros import debian
 
-class Distro(fedora.Distro):
+class Distro(debian.Distro):
     pass
 
 # vi: ts=4 expandtab
